worker/src/Channel/UnixStreamSocket.cpp

```diff
@@ -292,7 +259,8 @@ namespace Channel
 				json jsonMessage = json::parse(jsonStart, jsonStart + jsonLen);
 
 				// Notify the listener.
-				this->listener->OnConsumerSocketMessage(this, jsonMessage);
+				if (this->listener)
+					this->listener->OnConsumerSocketMessage(this, jsonMessage);
```

Comment:
Wh is this `if (this->listener)` needed? We always set the `listener` before using the class methods. Otherwise it would be a bug.

Reply:
In my opinion, the `listener` is not a necessary one, it means without any listener this `UnixStreamSocket` should work! If not so, maybe design the `listener` as a reference to the listener could be a better implementation. Could you agree with me?